compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.td

```diff
@@ -1381,6 +1383,7 @@ def FLOW_TensorEmptyOp : FLOW_PureOp<"tensor.empty", [
 
   let extraClassDeclaration = [{
     // StreamableOpInterface:
+    bool isMetadata() { return true; }
```

Comment:
hmm empty/clone aren't metadata-only, as they result in allocations (empty) and device operations (copies). we could add an isTransfer for these kind of ops that aren't compute and aren't metadata that we want to allow to be hoisted.